plugins/la_spec.py

```diff
@@ -166,29 +166,38 @@ class LASEMSpectrumMDStream(SEMSpectrumMDStream):
     """
 
     def estimateAcquisitionTime(self):
-        total_time = super(LASEMSpectrumMDStream, self).estimateAcquisitionTime()
+        total_time = super().estimateAcquisitionTime()
 
         # Add time to move the lens 2
         if self._sccd.l2:
             total_time += LENS2_MOVE_TIME
 
         return total_time
 
-    def _adjustHardwareSettings(self):
-        # Ideally this would be done in .acquire(), just after .prepare(), or in .prepare()
-        # But as they return futures, it's a little more complex to do than just
-        # here.
+    # Same as LASpectrumSettingsStream: override ._prepare_opm() to move the axes
+    # *after* the standard optical path management is complete.
+    def _prepare_opm(self):
+        # Return a future which calls the OPM _and_ updates the "special" axes
+        f = futures.Future()
+        executeAsyncTask(f, self._set_optical_path)
+        return f
+
+    def _set_optical_path(self):
+        f = super()._prepare_opm()
+        f.result()
+        # Take care of the axes as soon as the OPM is done
+        # Note: it's sub-optimal, as the OPM will explicitly move the axes away
+        # while we maybe end-up putting them back.
```

Comment:
I am not sure if adding the done callback will be useful or not
`def _set_optical_path(self):
    # Run the parent’s OPM preparation asynchronously
    f = super()._prepare_opm()
    
    # When the future completes, handle the axes
    f.add_done_callback(self._handle_axes_after_opm)

def _handle_axes_after_opm(self, future):
    # This method runs when the OPM future is done
    future.result()  # This gets the result of the OPM call (now complete)
    self._sccd._changeLensAxes()  # Adjust the axes afterward
`

Reply:
Ideally I would write it the way you proposed, however, the problem is that _prepare_opm() has to return a Future which is *completed* when all the axes are at the right place, because immediately after the acquisition starts. With your suggestion, the returned Future would be completed as soon as the movements are started. That's not good.